main.cpp

```diff
@@ -56,46 +55,43 @@ struct FloatType
         value = nullptr;
     }
 
-    FloatType& add(float f) 
-    { 
-        *value += f;
-        return *this; 
-    }
-    FloatType& subtract(float f) 
-    { 
-        *value -= f;
-        return *this; 
-    }
-    FloatType& multiply(float f) 
-    { 
-        *value *= f;
-        return *this; 
-    }
-    FloatType& divide(float f) 
-    { 
-        if(f == 0) { std::cout << "Warning: dividing by 0 results in undefined value!" << std::endl; }
-        *value /= f;
-        return *this; 
-    }
+    FloatType& add(float f);
+    FloatType& subtract(float f);
+    FloatType& multiply(float f);
+    FloatType& divide(float f);
 
-    // add/subtract/multiply/divide IntType 
+    // Not sure how to handle the potential 'else' case here
+    operator float() { if(value) return *value; }
```

Comment:
you know `value` is not a `nullptr` because of how the class is constructed. 
- [ ] remove the check to clear the warning.